mvlearn/decomposition/groupica.py

```diff
@@ -0,0 +1,276 @@
+        multiple_outputs=True,
+        prewhiten=False,
+        solver="picard",
+        ica_kwargs={},
+        random_state=None,
+    ):
+        if solver not in ["picard", "fastica"]:
+            raise ValueError(
+                "Invalid solver, must be either `fastica` or `picard`"
+            )
+        self.n_components = n_components
+        self.n_individual_components = n_individual_components
+        self.multiple_outputs = multiple_outputs
+        self.prewhiten = prewhiten
+        self.solver = solver
+        self.ica_kwargs = ica_kwargs
+        self.random_state = random_state
+
+    def fit(self, Xs, y=None):
+        r"""Fit  to the data and transform the data.
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+             - Xs length: n_views
+             - Xs[i] shape: (n_samples, n_features_i)
+        y : array, shape (n_samples,), optional
+
+        Returns
+        -------
+        sources : array, shape (n_samples, n_components)
+             Estimated sources
+        """
+        Xs = check_Xs(Xs, copy=True)
+        self.means_ = [np.mean(X, axis=0) for X in Xs]
+        gpca = GroupPCA(
+            n_components=self.n_components,
+            n_individual_components=self.n_individual_components,
+            prewhiten=self.prewhiten,
+            whiten=True,
+            multiple_outputs=False,
+            random_state=self.random_state,
+        )
+        X_pca = gpca.fit_transform(Xs)
+        self.grouppca_ = gpca
+        if self.solver == "fastica":
+            K, W, sources = fastica(
+                X_pca, **self.ica_kwargs, random_state=self.random_state
+            )
+        else:
+            K, W, sources = picard(
+                X_pca.T, **self.ica_kwargs, random_state=self.random_state
+            )
+            sources = sources.T
+        if K is not None:
+            self.components_ = np.dot(W, K)
+        else:
+            self.components_ = W
+        self.mixing_ = linalg.pinv(self.components_)
+        # Compute individual unmixing matrices by least-squares
+        self.individual_mixing_ = []
+        self.individual_components_ = []
+        sources_pinv = linalg.pinv(sources)
+        for X, mean in zip(Xs, self.means_):
+            lstq_solution = np.dot(sources_pinv, X - mean)
+            self.individual_components_.append(linalg.pinv(lstq_solution).T)
+            self.individual_mixing_.append(lstq_solution.T)
+        self.n_components_ = gpca.n_components_
+        self.n_features_ = gpca.n_features_
+        self.n_samples_ = gpca.n_samples_
+        self.n_views_ = gpca.n_views_
+        return self
+
+    def transform(self, Xs, y=None):
+        r"""
+        Transform the data Xs into sources.
+
+        Parameters
+        ----------
+        Xs: list of array-likes
+            - Xs shape: (n_views,)
+            - Xs[i] shape: (n_samples, n_features_i)
+        y : array, shape (n_samples,), optional
```

Comment:
I slightly prefer a description like `y : ignored, included for API compliance.` so there's no confusion